knapsack/branch_and_bound/README.md

```diff
@@ -0,0 +1,59 @@
+# Least Cost Branch and Bound algorithm to solve 0-1 Knapsack Problem
+
+The overview is taken from:
+https://en.wikipedia.org/wiki/Branch_and_bound
+https://www.geeksforgeeks.org/0-1-knapsack-using-least-count-branch-and-bound/#:~:text=Branch%20and%20Bound%20can%20be,one%20with%20the%20least%20cost.
+
+---
+
+## Overview
+
+Branch and bound (BB, B&B, or BnB) is an algorithm design paradigm for discrete and combinatorial optimization problems, as well as mathematical optimization. A branch-and-bound algorithm consists of a systematic enumeration of candidate solutions by means of state space search: the set of candidate solutions is thought of as forming a rooted tree with the full set at the root. The algorithm explores branches of this tree, which represent subsets of the solution set. Before enumerating the candidate solutions of a branch, the branch is checked against upper and lower estimated bounds on the optimal solution, and is discarded if it cannot produce a better solution than the best one found so far by the algorithm.
+
+The least cost(LC) is considered the most intelligent as it selects the next node based on a Heuristic Cost Function. It picks the one with the least cost.
+As 0/1 Knapsack is about maximizing the total value, we cannot directly use the LC Branch and Bound technique to solve this. Instead, we convert this into a minimization problem by taking negative of the given values.
+Follow the steps below to solve the problem:
+
+1. Sort the items based on their value/weight(V/W) ratio.
```

Comment:
It'd be helpful if we make it \`V\` and \`W\`. Same for all other variables like minLB -> \`minLB\` -> `minLB`